BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -98,27 +89,41 @@ extension ElementContent {
         init(layout: LayoutType) {
             self.layout = layout
         }
+        
+        /// Adds the given child element.
+        public mutating func add(element: Element, traits: LayoutType.Traits = LayoutType.defaultTraits, key: AnyHashable? = nil) {
+            let child = Child(
+                element: element,
+                content: element.content,
+                traits: traits,
+                key: key
+            )
+            
+            children.append(child)
+        }
+        
+        fileprivate struct Child : Measurable {
 
-    }
-
+            var element: Element
+            var content: ElementContent
+            var traits: LayoutType.Traits
+            var key: AnyHashable?
 
+            func measure(in constraint: SizeConstraint) -> CGSize {
+                content.measure(in: constraint)
+            }
+        }
+    }
 }
 
-extension ElementContent.Builder {
-
-    /// Adds the given child element.
-    public mutating func add(traits: LayoutType.Traits = LayoutType.defaultTraits, key: AnyHashable? = nil, element: Element) {
-        let child = Child(
-            traits: traits,
-            key: key,
-            content: element.content,
-            element: element)
-        children.append(child)
-    }
 
+fileprivate protocol AnyContentStorage : Measurable {
+    var childCount: Int { get }
+    func performLayout(attributes: LayoutAttributes) -> [(identifier: ElementIdentifier, node: LayoutResultNode)]
 }
 
-extension ElementContent.Builder: ContentStorage {
+
+extension ElementContent.ContentStorage : AnyContentStorage {
```

Comment:
Hmm. To me, this implementation feels like its distinguishing feature is that you can use it to build the list of its children. How about `BuildingContentStorage`? Or maybe `DefaultContentStorage`, since it's currently the only type?